newrelic/internal/transform/metric.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package transform
+
+import (
+	"errors"
+
+	"github.com/newrelic/newrelic-telemetry-sdk-go/telemetry"
+	"go.opentelemetry.io/otel/api/core"
+	"go.opentelemetry.io/otel/api/metric"
+	metricsdk "go.opentelemetry.io/otel/sdk/export/metric"
+	"go.opentelemetry.io/otel/sdk/export/metric/aggregator"
+)
+
+// ErrUnimplementedAgg is returned when a transformation of an unimplemented
+// aggregator is attempted.
+var ErrUnimplementedAgg = errors.New("unimplemented aggregator")
+
+// Record transforms an OpenTelemetry Record into a Metric.
+//
+// An ErrUnimplementedAgg error is returned for unimplemented Aggregators.
+func Record(service string, r metricsdk.Record) (telemetry.Metric, error) {
+	desc := r.Descriptor()
+	attrs := attributes(service, desc, r.Labels())
+	switch a := r.Aggregator().(type) {
+	case aggregator.MinMaxSumCount:
+		return minMaxSumCount(desc, attrs, a)
+	case aggregator.Sum:
+		return sum(desc, attrs, a)
+	}
+	return nil, ErrUnimplementedAgg
```

Comment:
What happens in otel when this error is returned? Do the metrics get retried next harvest period?

Reply:
That's ultimately going to be up to the configured Controller. This error will get rolled up all the way back to the caller of the `Export` method this exporter implements on the `Exporter` interface. This method [is called by the Controller](https://github.com/open-telemetry/opentelemetry-go/blob/master/sdk/metric/controller/push/push.go#L181) in the OpenTelemetry Go SDK. The default "push" Controller (currently the only controller) has a configurable error handler function that [by default prints the error to STDERR](https://github.com/open-telemetry/opentelemetry-go/blob/3008c1bf025b4710f3760c974ec679ae658e0d76/sdk/metric/sdk.go#L326).